nh_clinical/patient.py

```diff
@@ -51,7 +51,34 @@ class NhClinicalPatient(osv.Model):
     ]
 
     @staticmethod
-    def _clean_identifiers(dirty_vals):
+    def _check_identifier_for_bad_chars(value):
+        """
+        Check for bad characters in string
+
+        :param value: string to check
+        """
+        unallowed_chars = '[^a-zA-Z0-9_\-\s]'
+        allowed_chars = re.sub(unallowed_chars, '', value)
+        if allowed_chars != value:
+            raise ValidationError(
+                'Patient identifier can only contain '
+                'alphanumeric characters, hyphens and underscores'
+            )
```

Comment:
Agree on second point. 

Reason for not borking on white space is that if there's just white space (like a trailing white space) we don't want to raise an error, we just want to remove it so they don't have to resubmit the form for a simple issue caused by a bad copy and paste jobby